src/Sylius/Bundle/AdminBundle/Resources/views/Product/Show/_header.html.twig

```diff
@@ -9,12 +9,42 @@
         </h1>
         {% include "@SyliusAdmin/Product/Show/_breadcrumb.html.twig" %}
     </div>
-    <div class="middle aligned column">
-        <div class="ui right floated buttons">
-            <a class="ui labeled icon button {% if not product.enabled %} disabled {% endif %}" href="{{ path('sylius_shop_product_show', {'slug': product.slug, '_locale': product.translation.locale}) }}">
-                <i class="angle right icon"></i>
-                {{ 'sylius.ui.show_product_in_shop_page'|trans }}
-            </a>
+    {% if product.channels|length > 1 %}
```

Comment:
You don't check if channels are enabled for this product, maybe you can try to use twig filter?
https://twig.symfony.com/doc/2.x/filters/filter.html